src/index.ts

```diff
@@ -17,14 +16,55 @@ interface IDeployAppArgs {
    * The origin of the command e.g. main-menu, context-menu, etc.
    */
   origin?: string;
+  /**
+   * Whether or not the deployment form should be opened in headless mode
+   */
+  headless?: boolean;
+}
+
+interface IPathWidget {
+  /**
+   * The context object associated with the widget.
+   * Contains the metadata and information about the widget.
+   */
+  context: {
+    /**
+     * The path of the current document or resource associated with the widget.
+     * Typically represents the file path or location of the notebook or document.
+     */
+    path: string;
+  };
 }
 
+const hasContextPath = (widget: any): widget is IPathWidget => {
+  return widget && widget.context && typeof widget.context.path === 'string';
+};
+
 const jhubAppsPlugin: JupyterFrontEndPlugin<void> = {
   id: 'jupyterlab-jhub-apps:commands',
   description: 'Adds additional commands used by jhub-apps.',
   autoStart: true,
   activate: (app: JupyterFrontEnd) => {
-    const openURL = (url: string) => {
+    const deploymentFormUrl = '/services/japps/create-app';
+
+    const openInIFrame = (deployUrl: string) => {
+      try {
+        const iframe = document.createElement('iframe');
+        iframe.src = deployUrl;
+        iframe.width = '100%';
+        iframe.height = '100%';
+        iframe.style.border = 'none';
```

Comment:
JupyterLab includes [`IFrame`](https://jupyterlab.readthedocs.io/en/stable/api/classes/apputils.IFrame-1.html?highlight=Iframe#) class; you do not have to use it but you it includes good security defaults (`sandbox` - some of the options may need disabling) and sets the width/height to 100% by default. In the next version if will also expose `lazy` `loading` attribute which is nice as it makes JupyterLab load faster when multiple tabs with iframes are open.